include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -0,0 +1,254 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__geometry_model_ellipsoidal_chunk_h
+#define __aspect__geometry_model_ellipsoidal_chunk_h
+
+#include <aspect/geometry_model/interface.h>
+#include <deal.II/grid/manifold.h>
+namespace
+{
+  using namespace dealii;
+  /**
+   * This geometry model implements an (3d) ellipsoidal chunk geometry which can be non-coordinate parallel.
+   * @author This plugin is a joined effort of Menno Fraters, D Sarah Stamps and Wolfgang Bangerth
+   */
+  namespace Transformations
+  {
+    using namespace dealii;
+
+    class EllipsoidalChunkGeometry : public ChartManifold<3,3>
+    {
+      public:
+        virtual
+        Point<3>
+        pull_back(const Point<3> &space_point) const;
+
+        virtual
+        Point<3>
+        push_forward(const Point<3> &chart_point) const;
+
+        Point<3>
+        rotate_rectangle_to_paralellegram(const Point<3> &x) const;
+
+        Point<3>
+        rotate_paralellegram(const Point<3> &x) const;
+
+      private:
+
+        static double    R;
+        static double    ellipticity;
+
+        Point<3> push_forward_ellipsoid (const Point<3> &phi_theta_d) const;
+        Point<3> pull_back_ellipsoid (const Point<3> &x) const;
+
+        Point<3> push_forward_topo (const Point<3> &phi_theta_d_hat) const;
+        Point<3> pull_back_topo (const Point<3> &phi_theta_d) const;
```

Comment:
you don't actually seem to use these last two functions. Can you remove them, and then just move the code of the `_ellipsoid` functions into the `push_forward` and `pull_back` functions?